### PR TITLE
analyzer: Add github release workflow.

### PR DESCRIPTION
**Pull request by hjpotter92**

### CODE REVIEW

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+            ${{ runner.os }}-go-
+      - name: Get dependencies
+        run: |
+          go mod download
+      - name: Build analyzer binary
+        run: |
+          mkdir -p releases/ build/
+          for platform in linux darwin windows
+          do
+            for arch in arm64 amd64
```
Comment (victorges) [inline .github/workflows/release.yaml:57]: Just an idea for the future, but would be nice to get this into a build matrix instead for building all versions in parallel 
https://docs.github.com/en/actions/using-workflows/advanced-workflow-features#using-a-build-matrix

**Review by victorges — APPROVED** [review]
LGTM!

Comments are more like questions or ideas for the future. I assume most of it is just how we are doing on go-livepeer and everywhere else, so we probably want to keep everything as similar as possible.

File: .github/workflows/release.yaml (hunk 1 above)
Comment (darkdarkdragon) [reply .github/workflows/release.yaml:57]: Matrix creates new job for every platform - which takes more time. With pure Go project there is no point to use them - Go perfectly cross-compiles to every platform. 
Using matrixes make sense when project uses CGO - so cross-compiling is not possible.

File: .github/workflows/release.yaml [hunk 2]
```diff
@@ -0,0 +1,79 @@
+name: Publish new release
+
+on:
+  push:
+    tags:
+      - "v*"
+
+jobs:
+  create_release:
+    name: Create Release
```
[the remark this answers is not in the document]
Comment (hjpotter92) [reply .github/workflows/release.yaml:10]: yeah, that struck me as odd too, but I followed the structure from go-livepeer.

**Pull request merged**